llvm/lib/Transforms/Instrumentation/SoftPointerAuth.cpp

```diff
@@ -259,8 +259,10 @@ bool SoftPointerAuth::runOnModule(Module &M) {
 
 Value *SoftPointerAuth::emitSign(IRBuilderTy &builder, Value *pointer,
                                  Value *key, Value *discriminator) {
-  auto call = builder.CreateCall(getSignPointerFn(),
-                                 { pointer, key, discriminator });
+  auto callee = getSignPointerFn();
+  auto call = builder.CreateCall(
```

Comment:
The get*Fn() helpers are only used by the emits, maybe you can change those to make them fit better, e.g., by using `FunctionCallee` for the variables and the helper return value?

Reply:
Also `getSignGenericFn` returns `SignPointerFn` instead of `SignGenericFn` looks like; that's gotta be a pretty old bug right there, lucky that sign_generic isn't very common ;)

Reply:
Well, mostly I think nobody is using this pass, but yeah, we should fix that bug.

Reply:
I've made `SignGenericFn` fix in a separate commit, unfortunately don't know how to test it. Can fold it into the main commit, if you think it is better.